### ISSUE TITLE
NullpointerException in 3.0.0-beta.3

### ISSUE THREAD

**Issue by mcsong** [barteksc/AndroidPdfViewer#459]
Hello, 
I got this error when some user click some links in the pdf file. 
-----------------
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'long java.lang.Long.longValue()' on a null object reference
       at com.shockwave.pdfium.PdfiumCore.getPageLinks(PdfiumCore.java:394)
       at com.github.barteksc.pdfviewer.PdfFile.getPageLinks(PdfFile.java:259)
       at com.github.barteksc.pdfviewer.DragPinchManager.checkLinkTapped(DragPinchManager.java:97)
       at com.github.barteksc.pdfviewer.DragPinchManager.onSingleTapConfirmed(DragPinchManager.java:68)
       at android.view.GestureDetector$GestureHandler.handleMessage(GestureDetector.java:312)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6682)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
------------------

**Comment by 1stmetro** [comment]
do you want to post the pdf ?

**Comment by mcsong** [comment]
Sorry, I don't have any pdf files to represent this error. It is collected by one of crash reporting service.

**Comment by 1stmetro** [comment]
The log appears to show a null, maybe the link is just a http:// and hence the link is null, maybe worth playing with links in pdf's to replicate the issue until you locate an issue as per your log, 

Catch the exception if the link is not valid.... etc... this would stop the crashing,

**Comment by mcsong** [comment]
As you can see, this code can be a null in PdfiumCore class in PdfiumAndroid project. 
//--------------------- 
long[] linkPtrs = nativeGetPageLinks(doc.mNativePagesPtr.get(pageIndex));
//---------------------

I think, doc.mNativePagesPtr.get(pageIndex) could be a null. 
So, I know how to fix this issue, and I would like to share this issue to you. 

Thank you,

**Comment by 1stmetro** [comment]
Post the code it may be added to the source

**Comment by mcsong** [comment]
//-------------------
List<PdfDocument.Link> links = new ArrayList<>();
long[] linkPtrs = nativeGetPageLinks(doc.mNativePagesPtr.get(pageIndex));
//-----

You can change from up to below.  

List<PdfDocument.Link> links = new ArrayList<>();
Long nativePagePtr = doc.mNativePagesPtr.get(pageIndex);
if(nativePagePtr == null) {
  return links;
}
long[] linkPtrs = nativeGetPageLinks(nativePagePtr);

Thank you,

**Comment by 1stmetro** [comment]
Create a pull request..

**Comment by mcsong** [comment]
I did a pull request to PdfiumAndroid. 
Thank you,

**Comment by sethchhim** [comment]
I tried using `3.0.0-beta.3` but get a crash similar to this issue. Here is my Play Store developer log:

```
  #00  pc 0000000000044308  /system/lib/libc.so (tgkill+12)
  #01  pc 0000000000041f09  /system/lib/libc.so (pthread_kill+32)
  #02  pc 000000000001ba1f  /system/lib/libc.so (raise+10)
  #03  pc 0000000000018bc1  /system/lib/libc.so (__libc_android_abort+34)
  #04  pc 0000000000016784  /system/lib/libc.so (abort+4)
  #05  pc 0000000000321729  /system/lib/libart.so (_ZN3art7Runtime5AbortEv+212)
  #06  pc 00000000000f3cad  /system/lib/libart.so (_ZN3art10LogMessageD2Ev+2212)
  #07  pc 000000000024f83d  /system/lib/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+1524)
  #08  pc 000000000024fc49  /system/lib/libart.so (_ZN3art9JavaVMExt9JniAbortFEPKcS2_z+68)
  #09  pc 0000000000276f6d  /system/lib/libart.so (_ZN3art3JNI11GetMethodIDEP7_JNIEnvP7_jclassPKcS6_+636)
  #10  pc 00000000000076a7  /data/app/com.sethchhim.viewerforubooquity-1/lib/arm/libjniPdfium.so (_ZN7_JNIEnv11GetMethodIDEP7_jclassPKcS3_+8)
  #11  pc 0000000000007db1  /data/app/com.sethchhim.viewerforubooquity-1/lib/arm/libjniPdfium.so (Java_com_shockwave_pdfium_PdfiumCore_nativeGetPageSizeByIndex+144)
  #12  pc 00000000008eb18b  /data/app/com.sethchhim.viewerforubooquity-1/oat/arm/base.odex
```

This is strange because the crash only happens after I build my project to apk. It does not crash while I'm using a debug build. For the time being I have reverted to `2.8.1` which works fine for now.

**Comment by barteksc** [comment]
Fixed in 3.0.0-beta.4
